### PR TITLE
Add `temp_bucket` parameter for resource google_dataproc_cluster (fixes #7927)

### LINKED ISSUE #7927: Temp bucket parameter for google_dataproc_cluster
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment. If the issue is assigned to the "modular-magician" user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If the issue is assigned to a user, that user is claiming responsibility for the issue. If the issue is assigned to "hashibot", a community member has claimed the issue already.

### Description

Dataproc cluster create api does have an option to provide --temp-bucket, would be nice to have this option via terraform
https://cloud.google.com/sdk/gcloud/reference/dataproc/clusters/create

User Case: Be able to provide a pre provisioned temp bucket to create a Dataproc cluster

### New or Affected Resource(s)

* google_dataproc_cluster

### PR DESCRIPTION
**Pull request by bartcode**
Allow using `temp_bucket` when creating a Dataproc cluster as addressed in #7927 .

Closes #7927 .

If this PR is for Terraform, I acknowledge that I have:
* [x]  Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
* [x]  [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
* [x]  Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/third_party/terraform/tests) (for handwritten resources or update tests).
* [x]  [Ran](https://github.com/hashicorp/terraform-provider-google/blob/master/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
* [x]  Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
dataproc: Added field `temp_bucket` to `google_dataproc_cluster` cluster config.
```

### CODE REVIEW

File: google/resource_dataproc_cluster_test.go [hunk 1]
```diff
@@ -419,6 +419,37 @@ func TestAccDataprocCluster_withStagingBucket(t *testing.T) {
...
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDataprocCluster_withTempBucketAndCluster(clusterName, bucketName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckDataprocClusterExists(t, "google_dataproc_cluster.with_bucket", &cluster),
+					resource.TestCheckResourceAttr("google_dataproc_cluster.with_bucket", "cluster_config.0.temp_bucket", bucketName)),
+			},
+			{
+				// Simulate destroy of cluster by removing it from definition,
+				// but leaving the storage bucket (should not be auto deleted)
```
Comment (melinath) [inline google/resource_dataproc_cluster_test.go:443]: Could you update this docstring to say "temp bucket"?

**Review by melinath — APPROVED** [review]
LGTM - seems pretty straightforward! I'll get started on upstreaming this.

File: google/resource_dataproc_cluster_test.go (hunk 1 above)
Comment (bartcode) [reply google/resource_dataproc_cluster_test.go]: Yes, done!

File: google/resource_dataproc_cluster.go [hunk 2]
```diff
@@ -158,6 +159,14 @@ func resourceDataprocCluster() *schema.Resource {
 							Description: ` The name of the cloud storage bucket ultimately used to house the staging data for the cluster. If staging_bucket is specified, it will contain this value, otherwise it will be the auto generated name.`,
 						},
 
+						"temp_bucket": {
+							Type:         schema.TypeString,
+							Optional:     true,
```
Comment (melinath) [inline google/resource_dataproc_cluster.go:164]:
```suggestion
							Optional:     true,
							Computed:     true,
```

**Review by melinath — CHANGES_REQUESTED** [review]
It looks like this ended up breaking all the DataprocCluster tests that _don't_ define an explicit temp bucket, since the server then assigns one for you.

The fix for this should be to mark the field as Computed as well as optional. That should still allow setting an explicit value while also accepting whatever the server sends back if there's not a value defined.

**Comment by bartcode** [comment]
Good find! I've updated it and also executed `TestAccDataprocCluster_singleNodeCluster`, `TestAccDataprocCluster_withStagingBucket`, and `TestAccDataprocCluster_withTempBucket` which all passed.

File: google/resource_dataproc_cluster.go (hunk 2 above)
Comment (bartcode) [reply google/resource_dataproc_cluster.go:164]: Good find! I've updated it and also executed TestAccDataprocCluster_singleNodeCluster, TestAccDataprocCluster_withStagingBucket, and TestAccDataprocCluster_withTempBucket which all passed.

Comment (melinath) [reply google/resource_dataproc_cluster.go:164]: great, glad to hear they pass locally now! I've upstreamed this change & the CI tests are running.

**Review by melinath — APPROVED** [review]
LGTM and tests are passing

**Comment by bartcode** [comment]
I noticed the comment in `magic-modules`, but it's been quiet for a while. Should I just wait or shall I remove the test that's suggested there? Your interpretation was correct, by the way, I copied some tests to ensure they also pass with this temporary bucket.

**Comment by melinath** [comment]
@bartcode thanks for checking back in. I lost track of this, sorry. Let me ping Sam.

**Review by melinath — CHANGES_REQUESTED** [review]
Okay, so there's just one thing to change - since this is a handwritten resource, the docs also need to be updated manually. They're in website/docs/r/. Thanks!

**Comment by bartcode** [comment]
Thanks! I wasn't aware of those docs, but they should be in order now!

**Review by melinath — APPROVED** [review]

**Pull request merged**